phoenix-core/src/main/java/org/apache/phoenix/coprocessor/AddColumnMutator.java

```diff
@@ -399,6 +401,19 @@ public MetaDataMutationResult validateAndAddMetadata(PTable table, byte[][] rowK
                                 rowKeyMetaData[TABLE_NAME_INDEX])));
             }
         }
+        if (isAddingColumns) {
+            //We're changing the application-facing schema by adding a column, so update the DDL
+            // timestamp
+            long serverTimestamp = EnvironmentEdgeManager.currentTimeMillis();
+            additionalTableMetadataMutations.add(MetaDataUtil.getLastDDLTimestampUpdate(tableHeaderRowKey,
+                clientTimeStamp, serverTimestamp));
+            for (PTable viewTable : childViews) {
```

Comment:
@ChinmaySKulkarni - I see how inheritance logic could work for PTables, but how would we make that work for the JDBC metadata API? (see PhoenixDatabaseMetaData) I'm asserting throughout my tests that PTables and the metadata API give the same answers, but the metadata API is, I believe, just querying the table or view header row in System.Catalog. 

(_Technically_, the JDBC metadata API is the "public" one and the PTables API is the "private" one, though in all the external Phoenix-based applications I work with for my day job, I've switched to using PTables because the JDBC metadata API is just too inefficient since it never caches results.)

Also just want to note that in a follow-up JIRA to this, we're going to need to (optionally) call to an external schema registry when we create a table/view or add/remove a column from a table/view, and that _will_ need to be synchronous, because the schema needs to be in the schema registry before DML using that schema starts being processed by the replication pipeline.

Reply:
Will make sure to exempt diverged columns, thanks for pointing that out.

Reply:
@ChinmaySKulkarni - is there an authoritative way to tell if a view is diverged? I don't see a PTable property. Is ViewUtil.isDivergedView(PTable) usable? Would I have to use ViewUtil.isDivergingView(PColumn, PTable) instead? Thanks!